test/solver_tests/snj_solver_test.py

```diff
@@ -0,0 +1,789 @@
+"""
+Test SpectralNeighborJoiningSolver in Cassiopeia.solver.
+"""
+import unittest
+from typing import List
+from unittest import mock
+import itertools
+import networkx as nx
+from networkx.classes.digraph import DiGraph
+import numpy as np
+import pandas as pd
+
+import cassiopeia as cas
+
+
+def find_triplet_structure(triplet, T):
+    """Identify the two nodes with the most similar ancestry.
+
+    Args:
+        triplet (str): name of nodes to check
+        T (tree): tree in which the nodes are in
+
+    Returns:
+        str: the two nodes that have the most similar number of
+            common ancestors.
+    """
+    a, b, c = triplet[0], triplet[1], triplet[2]
+    a_ancestors = [node for node in nx.ancestors(T, a)]
+    b_ancestors = [node for node in nx.ancestors(T, b)]
+    c_ancestors = [node for node in nx.ancestors(T, c)]
+    ab_common = len(set(a_ancestors) & set(b_ancestors))
+    ac_common = len(set(a_ancestors) & set(c_ancestors))
+    bc_common = len(set(b_ancestors) & set(c_ancestors))
+    structure = "-"
+    if ab_common > bc_common and ab_common > ac_common:
+        structure = "ab"
+    elif ac_common > bc_common and ac_common > ab_common:
+        structure = "ac"
+    elif bc_common > ab_common and bc_common > ac_common:
+        structure = "bc"
+    return structure
+
+
+def assertTripletCorrectness(
+    self, nodes: List[str], expected_tree: DiGraph, observed_tree: DiGraph
+):
+    """Checks whether two trees are identical regardless of internal node
+    names.
```

Comment:
"Checks if two trees are identical regardless of internal node names.", just to condense to 1 line?